htlcswitch/queue.go

```diff
@@ -137,10 +148,12 @@ func (p *packetQueue) packetCoordinator() {
 				atomic.AddInt64(&p.totalHtlcAmt, int64(-nextPkt.amount))
 				p.queueCond.L.Unlock()
 			case <-p.quit:
+				atomic.StoreInt32(&p.streamShutdown, 1)
 				return
 			}
 
 		case <-p.quit:
+			atomic.StoreInt32(&p.streamShutdown, 1)
```

Comment:
would it make a difference to let the `quit` case be the first case in all the selects?

Reply:
Nope, the select case is selected randomly.